File: 12-Searching-Algorithms/02-Binary-Search/binary_search.py
n = int(input("Enter number of elements: "))
arr = list(map(int, input("Enter sorted elements: ").split()))

target = int(input("Enter element to search: "))

low = 0
high = n - 1
found = -1

while low <= high:
    mid = low + (high - low) // 2

    if arr[mid] == target:
        found = mid
        break

    elif arr[mid] < target:
        low = mid + 1

    else:
        high = mid - 1

if found != -1:
    print("Element found at index:", found)
else:
    print("Element not found.")

File: 12-Searching-Algorithms/02-Binary-Search/binary_search.c
#include <stdio.h>

#define MAX 100

int main() {
    int arr[MAX], n, target;
    int low, high, mid;
    int found = -1;

    printf("Enter number of elements: ");
    scanf("%d", &n);

    printf("Enter sorted elements: ");

    for(int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    printf("Enter element to search: ");
    scanf("%d", &target);

    low = 0;
    high = n - 1;

    while(low <= high) {
        mid = low + (high - low) / 2;

        if(arr[mid] == target) {
            found = mid;
            break;
        }

        if(arr[mid] < target)
            low = mid + 1;
        else
            high = mid - 1;
    }

    if(found != -1)
        printf("Element found at index: %d\n", found);
    else
        printf("Element not found.\n");

    return 0;
}